### PR TITLE
Allow phrases to contain more than nine numeric placeholders

### LINKED ISSUE #1374: \Magento\Framework\Phrase\Renderer\Placeholder::render() can't handle more than 9 placeholders
Using the code fragment

```
<?= 
__(
    '%1<br>%2<br>%3<br>%4<br>%5<br>%6<br>%7<br>%8<br>%9<br>%10',
    'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'
); 
?>
```

in a phtml file will result in the following **wrong** output:

```
one
two
three
four
five
six
seven
eight
nine
one0
```

because Placeholder Renderer uses _str_replace_ which replaces %10 in the first place.

__Question:__ What exactly was wrong with the old sprintf rendering?!

### PR DESCRIPTION
**Pull request by joshdifabio**
Fixes #1374

### CODE REVIEW

**Comment by kandy** [comment]
better to use `strtr` instead of `str_replace`to resolve issue http://3v4l.org/LKT3k

**Comment by joshdifabio** [comment]
> better to use strtr instead of str_replaceto resolve issue http://3v4l.org/LKT3k

Good call :). Updated the PR.

File: lib/internal/Magento/Framework/Phrase/Renderer/Placeholder.php [hunk 1]
```diff
@@ -23,13 +23,16 @@ public function render(array $source, array $arguments)
...
-            $text = str_replace($placeholders, $arguments, $text);
+            $placeholders = array_map([$this, 'keyToPlaceholder'], array_keys($arguments));
+            $pairs = array_combine($placeholders, $arguments);
+            $text = strtr($text, $pairs);
         }
 
         return $text;
     }
+
+    private function keyToPlaceholder($key)
```
Comment (orlangur) [inline lib/internal/Magento/Framework/Phrase/Renderer/Placeholder.php]: Why not just make it a closure?

Comment (joshdifabio) [inline lib/internal/Magento/Framework/Phrase/Renderer/Placeholder.php]: I did that initially, but the anonymous function didn't inherit any variables from its parent scope (no *use* directive) so it seemed like a needless performance overhead to instantiate a `Closure` every time `render()` is called.

**Pull request merged**